src/com/facebook/buck/android/MergeAndroidResourcesStep.java

```diff
@@ -216,29 +236,55 @@ public StepExecutionResult execute(ExecutionContext context)
               filesystem,
               useOldStyleableFormat);
 
-      // If a resource_union_package was specified, copy all resource into that package,
-      // unless they are already present.
-      if (unionPackage.isPresent()) {
+      ImmutableSet.Builder<String> requiredPackages = ImmutableSet.<String>builder();
+
+      if (skipPrebuiltRDotJava) {
+        // If a resource_union_package was specified and skip_prebuilt_r_dot_java is true,
+        // copy all resource into that package and remove all other packages.
+        Preconditions.checkArgument(
+            unionPackage.isPresent(),
+            "union_package should be specified if skip_prebuilt_r_dot_java is set");
+
         String unionPackageName = unionPackage.get();
-        // Create a temporary list to avoid concurrent modification problems.
```

Comment:
The change loses this comment, and it's an important one

Reply:
added the comment back